runtime/vm/ModularityHashTables.c

```diff
@@ -189,3 +211,54 @@ findModuleInfoForModule(J9VMThread *currentThread, J9ClassLoader *classLoader, J
 	targetPtr = (J9ModuleExtraInfo *)hashTableFind(classLoader->moduleExtraInfoHashTable, (void *)&moduleInfo);
 	return targetPtr;
 }
+
+BOOLEAN
+didPackageExistInJAVA8(J9VMThread *currentThread, J9UTF8 *pkgName)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	BOOLEAN success = TRUE;
+	BOOLEAN result = FALSE;
+
+	/* check to see if we have loaded the java8 packages table */
+	if (J9_ARE_NO_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_JAVA8_PACKAGES_TABLE_LOADED)) {
+		/* lazily add a new table because we may not always need it, it is only required in
+		 * cases where an un-named module attempts to reflectively access non-exported package
+		 * that existed in java8
+		 */
+		omrthread_monitor_enter(vm->java8PackagesTableMutex);
+		if (J9_ARE_NO_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_JAVA8_PACKAGES_TABLE_LOADED)) {
```

Comment:
Have you seen https://github.com/eclipse/openj9/blob/master/runtime/bcutil/attrlookup.gperf ? We use `gperf` to create a perfect hash which making lookups faster / cheaper.